### ISSUE TITLE
Semver check regex does not recognize Home-Assistant Beta versions

### ISSUE THREAD

**Issue by brendann993** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#238]
### Checklist

- [X] I have updated the card to the latest version available
- [X] I have cleared the cache of my browser
- [X] I have checked if the problem is already reported

### The problem

The regex that checks Home-Assistant version does not expect a beta version at the end and hence fails for example on version 2021.12.0b1
Updating it from
```
Wi = /^v?(\d+)(?:\.([x*]|\d+)(?:\.([x*]|\d+)(?:\.([x*]|\d+))?(?:-([\da-z\-]+(?:\.[\da-z\-]+)*))?(?:\+[\da-z\-]+(?:\.[\da-z\-]+)*)?)?)?$/i
```
to
``` 
Wi = /^v?(\d+)(?:\.([x*]|\d+)(?:\.([x*]|\d+)(?:\.([x*]|\d+))?(?:-([\da-z\-]+(?:\.[\da-z\-]+)*))?(?:\+[\da-z\-]+(?:\.[\da-z\-]+)*)?)?)?\w*$/i
allows the characters at the end so the string matches but does not capture the beta version.
```

### What version of a card has described problem?

2.0.7

### What vacuum model do you have problems with?

n/a

### Which integration do you use to control your vacuum (link)?

https://www.home-assistant.io/integrations/xiaomi_miio/

### What browser (browsers/apps) does have this problem?

All

### What version of Home Assistant do you use?

2021.12.0b1

### What type of installation are you running?

Home Assistant OS

### Javascript errors shown in the browser's console (if applicable)

```shell
Uncaught (in promise) Error: Invalid argument not valid semver ('2021.12.0b1' received)
    at Yi (xiaomi-vacuum-map-card.js?hacstag=193372044207:53:67625)
    at xiaomi-vacuum-map-card.js?hacstag=193372044207:53:67232
    at Gi (xiaomi-vacuum-map-card.js?hacstag=193372044207:53:67340)
    at Function.usesSensors (xiaomi-vacuum-map-card.js?hacstag=193372044207:53:69004)
    at Function.generate (xiaomi-vacuum-map-card.js?hacstag=193372044207:557:5407)
    at bn._setPresetIndex (xiaomi-vacuum-map-card.js?hacstag=193372044207:836:11759)
    at bn._firstHass (xiaomi-vacuum-map-card.js?hacstag=193372044207:836:10432)
    at bn.set hass [as hass] (xiaomi-vacuum-map-card.js?hacstag=193372044207:836:8657)
    at i.value (hui-masonry-view.ts:44:1)
    at hui-masonry-view.ts:44:1
```

### Additional information

While running beta is a edge case that doesn't really need to be supported it should still function for testing

**Comment by PiotrMachowski** [comment]
This regex is a part of `compare-versions` library, so it can't be adjusted this way. I will create a different workaround for it 👍

**Comment by brendann993** [comment]
Ah yeah I was looking only as the dist rather than the source 👍🏻

**Comment by PiotrMachowski** [comment]
fixed in `v2.0.8`
